### PR TITLE
Fix for params chain

### PR DESCRIPTION
**Pull request by info-rchitect**

### CODE REVIEW

**Comment by ginty** [comment]
What was wrong with the original version?

I don't see how the behavior of the original vs. this would be any different...

**Comment by info-rchitect** [comment]
> What was wrong with the original version?
> I don't see how the behavior of the original vs. this would be any different...

Try running the last spec test with the old version

File: lib/origen/parameters/set.rb [hunk 1]
```diff
@@ -61,11 +61,12 @@ def method_missing(method, *args, &block)
             self[m] = args.first
           end
         else
+          # binding.pry if $stop == 1
           if args.length != 0
             super
           else
             if !key?(method)
-              super
+              nil
```
Comment (priyavadan) [inline lib/origen/parameters/set.rb]: @info-rchitect not sure I understand the need for this?

Comment (info-rchitect) [reply lib/origen/parameters/set.rb]: Without that change this spec test will fail:

~~~ruby
ip.params(:not_chain).chain.should == nil
~~~

Comment (priyavadan) [reply lib/origen/parameters/set.rb]: if I understand correctly, it'll also effect the min and max HASH methods?

ie ip.params(:it_is_a_range).max will return a nil? I could be wrong but i just want to understand

Comment (info-rchitect) [reply lib/origen/parameters/set.rb]: let me check that test case

Comment (info-rchitect) [reply lib/origen/parameters/set.rb]: I updated the code and the spec tests.

Comment (priyavadan) [reply lib/origen/parameters/set.rb]: @info-rchitect thanks, it covers all the possible uses cases we have internally, looks good to me.

@ginty thoughts?

**Pull request merged**